### PR TITLE
Generic point editor control

### PR DESCRIPTION
**Pull request by CartBlanche**
Refactor to support multiple 2 point objects.

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/PointEditorControl.cs [hunk 1]
```diff
@@ -9,77 +9,31 @@
 
 namespace Xamarin.PropertyEditing.Mac
 {
-	internal class PointEditorControl : PropertyEditorControl
+	internal class PointEditorControl : BasePointEditorControl<CGPoint>
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/PointEditorControl.cs]: If we have multiple subclasses that serve as shells for particular implementing types (`CGPoint`), we should name them accordingly.

File: Xamarin.PropertyEditing.Mac/Controls/SizeEditorControl.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+	internal class SizeEditorControl : BasePointEditorControl<CGSize>
+	{
+		public SizeEditorControl ()
+		{
+			XLabel.Frame = new CGRect (0, 0, 40, 24);
+			XLabel.StringValue = "Width:"; // TODO Localise
+
+			XEditor.Frame = new CGRect (45, 0, 50, 20);
+			XEditor.Activated += (sender, e) => {
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/SizeEditorControl.cs]: We should consider the hooks in the base class that calls a virtual method to update the value as we should _also_ be updating value on blur.

File: Xamarin.PropertyEditing.Mac/Controls/PointEditorControl.cs (hunk 1 above)
Comment (CartBlanche) [inline Xamarin.PropertyEditing.Mac/Controls/PointEditorControl.cs]: So following a CGPointPointEditorControl and CGSizeEditorControl in this case?

File: Xamarin.PropertyEditing.Mac/Controls/SizeEditorControl.cs (hunk 2 above)
Comment (CartBlanche) [inline Xamarin.PropertyEditing.Mac/Controls/SizeEditorControl.cs]: Can you clarify what you mean by 
> on blur
?

Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/SizeEditorControl.cs]: Loses focus

**Pull request closed without merging**